File: final-project-backend-resumaid/src/main/java/nus/iss/edu/sg/final_project_backend_resumaid/model/Resume.java
package nus.iss.edu.sg.final_project_backend_resumaid.model;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "resume")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Resume {

    @Id
    private String id;

    private String userId;
    private String title;
    private String fullName;
    private String phone;
    private String email;
    private List<Education> education;
    private List<Work> work;
    private List<Cca> cca;
    private String additional;
    private String url;
    private Date creationTime;
    private Date lastUpdateTime;

}

File: final-project-frontend-resumaid/src/app/models/resume.ts
export interface ResumeSlice {
    resumeForm: any
}

export interface education {

    eName: string
    eCountry: string
    eCert: string
    eFrom: Date
    eCurrent: boolean
    eTo: Date
    ePoints: string
}

export interface work {

    wName: string
    wCountry: string
    wRole: string
    wFrom: Date
    wCurrent: boolean
    wTo: Date
    wPoints: string
}

export interface cca {
    cName: string
    cCountry: string
    cTitle: string
    cFrom: Date
    cCurrent: boolean
    cTo: Date
    cPoints: string
}

export interface resume {

    id: string
    userId: string
    title: string
    fullName: string
    phone: string
    email: string
    education: education[]
    work: work[]
    cca: cca[]
    additional: string
    url: string
    creationTime: Date
    lastUpdateTime: Date
}
